packages/strapi-design-system/src/Table/Table.js

```diff
@@ -2,96 +2,103 @@ import React, { useEffect, useRef, useState } from 'react';
 import PropTypes from 'prop-types';
 import { TableContext } from './TableContext';
 import { KeyboardKeys } from '../helpers/keyboardKeys';
+import { focusFocusable } from './focusFocusable';
 
 export const Table = ({ colCount, rowCount, jumpStep, ...props }) => {
   const tableRef = useRef(null);
+  const mountedRef = useRef(false);
   /**
    * Rows will always have n+1 line because of the <tr><th></th></tr> elements
    */
   const [rowIndex, setRowIndex] = useState(0);
   const [colIndex, setColIndex] = useState(0);
 
   useEffect(() => {
-    const handleKeyDown = (e) => {
-      switch (e.key) {
-        case KeyboardKeys.ARROW_RIGHT: {
-          e.preventDefault();
-          setColIndex((prevColIndex) => (prevColIndex < colCount - 1 ? prevColIndex + 1 : prevColIndex));
+    if (mountedRef.current) {
+      focusFocusable(tableRef.current);
+    }
+
+    if (!mountedRef.current) {
+      mountedRef.current = true;
```

Comment:
Don't you need the cleanup?

`return () => (mountedRef.current = false)`

Reply:
Do you think I need it? 🤔